### ISSUE TITLE
Connection Time To Live - Cannot set with default

### ISSUE THREAD

**Issue by etspaceman** [aws/aws-sdk-java-v2#1976]
If you attempt to configure an HTTP Client with Duration.ZERO as the value for `connectionTimeToLive`, you will receive an error:

`java.lang.IllegalArgumentException: connectionTimeToLive must be positive`

The default value for this configuration is Duration.ZERO, so this exception is inconsistent:

https://github.com/aws/aws-sdk-java-v2/blob/master/http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java#L119
 

## Expected Behavior
I expect to be able to pass in the default value here and not receive an exception.

## Current Behavior
I receive an exception

## Steps to Reproduce
```
NettyNioAsyncHttpClient.builder().connectionTimeToLive(Duration.ZERO).build()
```

## Possible Solution
The solution should be to include 0 as a viable option for this configuration

## Context
I was attempting to build a configurator for the builder, which submits defaults if a configuration is not otherwise provided. 

## Your Environment

* AWS Java SDK version used: 2.13.25
* JDK version used: 1.8
* Operating System and version: MacOS Mojave

**Comment by zoewangg** [comment]
Makes sense. Thanks for reporting the issue!

**Comment by zoewangg** [comment]
The fix #1978 has been merged and will be available in the next release.
